SMTP.Dockerfile

```
@@ -1,20 +1,36 @@
-# Use the official Rust image as a base image
-FROM rust:latest
+# Use the official Rust image as a base image for building
+FROM rust:latest AS builder
 
 # Set the working directory inside the container
 WORKDIR /app
 
 # Clone the GitHub repository
-RUN git clone https://github.com/zkemail/relayer-smtp.git
+RUN git clone https://github.com/zkfriendly/relayer-smtp.git
 
 # Change to the directory of the cloned repository
 WORKDIR /app/relayer-smtp
 
 # Build the Rust package
-RUN cargo build
+RUN cargo build --release
+
+# Use a smaller base image for the final stage
+FROM debian:bookworm-slim
+
+# Install necessary runtime dependencies and setup SSL certificates
+RUN apt-get update && apt-get install -y \
+    libssl3 \
+    ca-certificates \
+    && rm -rf /var/lib/apt/lists/* \
```

Missing curl dependency for health check
```suggestion
    curl \
    && rm -rf /var/lib/apt/lists/* \
```